src/actions/index.js

```diff
@@ -1,21 +1,21 @@
 //Actions
-
+import denverWeather from './'
 import fetch from 'isomorphic-fetch';
 export const RECEIVE_FORECAST = 'RECEIVE_FORECAST';
 export const REQUEST_FORECAST = 'REQUEST_FORECAST';
 const API_URL = 'http://api.openweathermap.org/data/2.5/weather?zip=94040,us&appid=b248d23bebc3f14165312ce58845f0c4'
 
-// export function fetchWeatherByZip(zip) {
-//   return dispatch => {
-//     dispatch({ type: REQUEST_FORECAST });
-//     return fetch(API_URL.replace(/zip/, zip))
-//       .then(req => req.json())
-//       .then(json => dispatch({
-//         type: RECEIVE_FORECAST,
-//         json
-//       }));
-//   };
-// }
+export function fetchWeatherByZip(zip) {
+  return dispatch => {
+    dispatch({ type: REQUEST_FORECAST });
+    return fetch(API_URL.replace(/zip/, zip))
```

Comment:
This is likely to be slow and a bit brittle. What if we used a function that returned the string for us?

For example:

```js
const apiUrl = (zip = '80202') => (
  `http://api.openweathermap.org/data/2.5/weather?zip=${zip},us&appid=b248d23bebc3f14165312ce58845f0c4`
);
```

Now, it's a super flexible function that returns a string that you can use.

```js
fetch(apiUrl(zip)).then(…)
```